src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php

```diff
@@ -520,36 +520,36 @@ public function get_item_schema() {
 					'readonly'    => true,
 				),
 				'editor_script'    => array(
-					'description' => __( 'Editor script handle.' ),
-					'type'        => array( 'string', 'null' ),
+					'description' => __( 'Editor script handles.' ),
+					'type'        => array( 'array', 'string', 'null' ),
```

Comment:
The same applies to all fields related to scripts and styles. Is extending the list of return types for the field in the REST API endpoint a breaking change? Or maybe updating the schema and releasing a dev note is enough?

Without this patch, it's possible to register multiple styles, but it's handled with a hook in WordPress core. In practice, that means only the first registered style gets exposed through REST API, and the rest is ignored. We could keep the same old, imperfect behavior if introducing an array would be a breaking change.

Reply:
While I have no special knowledge or experience to offer, in my own experience with the REST API, I think I would consider adding `array` to these properties to be a breaking change. It might still be an acceptable breaking change to make with a dev note, and there might even be precedent for adding `array` to properties before.

I think it would be unlikely that my (hypothetical) integration with this endpoint would be set up to handle an array correctly when `string|null` was expected until now. Would it work to create "plural" versions of these properties (`editor_scripts`, `scripts`, etc.) that support arrays with the intention of deprecating the existing "singular" properties (see https://core.trac.wordpress.org/ticket/52629)?

Reply:
@dlh01, thank you for sharing your feedback. 

>  I think it would be unlikely that my (hypothetical) integration with this endpoint would be set up to handle an array correctly when string|null was expected until now. 

This is exactly what I'm worried about here. However, the same in theory would apply to `WP_Block_Type` and its properties here. Introducing new properties would resolve some of the issues, but it might also create another set of issues where we might need to keep singular and plural versions in sync. Unless, we use magic methods like `__get` and `__set` for plural fields that operate on the first item in the array.

Reply:
The current way to deprecate REST API fields in the response looks as follows:

https://github.com/WordPress/wordpress-develop/blob/1d456706549ccf4f0cc867b8a17d1dceaabf8a77/src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php#L1430-L1461

Reply:
+1 for introducing a new field and deprecating the old one – the existing plugins will probably not handle the new values correctly. FYI @spacedmonkey @TimothyBJacobs 

As for the deprecation, I think the description method you suggested works nicely.

Reply:
Yeah, changing the response shape would be a breaking change. We can transparently make updates more liberal in what we accept, but this would change how a client can consume the resource.

I agree with @adamziel that coming up with a new property name is probably the best course of action. The REST API isn't really at all set up yet for versioning endpoints.

Also, I do think the same care should apply to the `WP_Block_Type` class. IMO it is of more critical importance too because it could cause server-side fatal errors.

> Unless, we use magic methods like __get and __set for plural fields that operate on the first item in the array.

Yep, I think this would be necessary. It is a problem with Core's architecture patterns of using public properties instead of getters and setters which could more easily be adapted to provide BC, but I digress.

Reply:
It looks like we are in agreement on the way to move forward, which ensures that we optimize for backward compatibility. My only concern now is how to name new fields, so we remove the ambiguity between the existing `styles` (Block style variations.) and `style` (Contains the handle that defines the block style.). We definitely can't rename `style` to `styles` 😞

Reply:
Can `handles` be used in the property name?

Reply:
Great thinking @TimothyBJacobs, it's exactly handles that we store there 👍🏻

Reply:
I have a working prototype with https://github.com/WordPress/wordpress-develop/pull/3108/commits/f48b1c11dea9cf82ad3f5e27af1bfef7641186a6. I still need to refactor the rest of the code to use new properties.

Reply:
Everything is ready for the final review.